src/snowflake/snowpark/_internal/ast_utils.py

```diff
@@ -46,6 +45,53 @@
 SRC_POSITION_TEST_MODE = False
 
 
+class ExtractAssignmentVisitor(ast.NodeVisitor):
+    def __init__(self) -> None:
+        super().__init__()
+        self.symbols: Optional[Union[str, List[str]]] = None
+
+    def visit_Assign(self, node) -> None:
+        assert len(node.targets) == 1
+        target = node.targets[0]
+
+        if isinstance(target, ast.Name):
+            self.symbols = target.id
+        elif isinstance(target, ast.Tuple):
+            self.symbols = [name.id for name in target.elts]
+        else:
+            raise ValueError(f"Unsupported target {ast.dump(target)}")
+
+
+def extract_assign_targets(source_line: str) -> Optional[Union[str, List[str]]]:
+    """
+    Extracts the targets as strings for a python assignment.
+    Args:
+        source_line: A string, e.g. "a, b, c = df.random_split([0.2, 0.3, 0.5])"
+
+    Returns:
+        None if extraction fails, or list of strings for the symbol names, or a single string if it is a single target.
+    """
+    try:
+        tree = ast.parse(source_line.strip())
+        v = ExtractAssignmentVisitor()
+        v.visit(tree)
+        return v.symbols
+    except Exception:
+        # It may happen that an incomplete source line is submitted.
+        # For this reason, try to parse one more time if '=' is found.
+        if "=" in source_line:
```

Comment:
Shouldn't this be a retry loop, or a pass that fixes up all potentially broken source lines? Or a pass that fixes up all source lines before attempting the tree traversal?

Reply:
I think only a single source line is given. In python to the best of my knowledge the left-side of an assign can't be broken up along multiple lines. Therefore, from an input parameter perspective we only need a single line. However, we could simplify by basically always searching for the first `'='` occurence and then using the substition with `None` (or some other generic token).

Do you think this would be a good refactor?

Python doesn't allow for `'='` as part of the name of an identifier, so we should be safe with this approach.

Reply:
You're right that would be a legal case. If we wanted to support this, we would have to do some eager line parsing till we succeed parsing something including an AST. Another problematic case may be

`df = df.select(...); df2 = df.select(...)`.

Here, the wrong symbol would be picked up for the second statement. I don't think we'll have a perfect solution without a lot of additional work. My suggestion is therefore to take up a more incremental approach. With this PR, the basic case of multiple targets is supported similar to the existing code. The existing code does not handle your example, not the multi-statement per-line case.

Do you think we could defer implementation for these more complex cases? The worst that will happen currently is that the symbol doesn't get captured and something like `res1, res2, ...` will get generated. Wdyt?

Reply:
We'll end up parsing Python for this at some point, and I agree it's not worth doing now. Let's go with a simple hack until someone breaks it.

Reply:
Sgtm, I want to pick up your code regarding the nesting here. I think when applying `source_line = source_line[: source_line.find("=")] + " = None"` directly, the code can be simplified. Will refactor and add a comment to explain.

Reply:
refactored.